modules/ding_nodelist/ding_nodelist.module

```diff
@@ -0,0 +1,2477 @@
+    '#weight' => -1,
+    '#parents' => array('more_links', $key, 'text'),
+    '#default_value' => (!empty($link['text'])) ? $link['text'] : '',
+  );
+
+  $form['links'] = array(
+    '#type' => 'textfield',
+    '#title' => t('URL'),
+    '#title_display' => 'invisible',
+    '#required' => FALSE,
+    '#description' => check_plain(t('Local path or external URL starting with http(s). Enter "<front>" to link to home page.')),
+    '#parents' => array('more_links', $key, 'links'),
+    '#default_value' => (!empty($link['links'])) ? $link['links'] : '',
+  );
+
+  return $form;
+}
+
+/**
+ * Builds filter form.
+ */
+function _ding_nodelist_taxonomy_filter($fields, $filter, $key) {
+  $form = array(
+    '#tree' => TRUE,
+    '#weight' => $key,
+  );
+  $form['filter_field'] = array(
+    '#type' => 'select',
+    '#title_display' => 'invisible',
+    '#options' => $fields,
+    '#default_value' => !is_null($filter) && !empty($filter['filter_field']) ? $filter['filter_field'] : '',
+    '#ajax' => array(
+      'wrapper' => 'ding_nodelist-taxonomy-filters',
+      'callback' => 'ding_nodelist_js_add_filter',
+      'method' => 'replace',
+      'effect' => 'fade',
+    ),
+    '#parents' => array('taxonomy_filters', $key, 'filter_field'),
+  );
+
+  if ($filter['term_reference'] == 'taxonomy_term_reference') {
+    $autocomplete = !is_null($filter) && !empty($filter['filter_field']) ? 'taxonomy/autocomplete/' . $filter['filter_field'] : NULL;
+  }
+  else {
+    $autocomplete = !is_null($filter) && !empty($filter['filter_field']) ? "entityreference/autocomplete/tags/" . $filter['filter_field'] . '/' . $filter['type'] . '/' . $filter['bundle'] . '/NULL' : NULL;
+  }
+
+  $form['filter_terms'] = array(
+    '#vocabulary_name' => isset($filter['vocabulary']) ? $filter['vocabulary'] : NULL,
+    '#reference_type' => $filter['term_reference'],
+    '#type' => 'textfield',
+    '#title_display' => 'invisible',
+    '#default_value' => !is_null($filter) && !empty($filter['filter_terms']) ? $filter['filter_terms'] : '',
+    '#parents' => array('taxonomy_filters', $key, 'filter_terms'),
+    '#maxlength' => 1024,
+    '#element_validate' => array('_ding_nodelist_taxonomy_autocomplete_validate'),
+    '#autocomplete_path' => $autocomplete,
+  );
+
+  $form['filter_delete_' . $key] = array(
+    '#name' => 'filter_delete_' . $key,
+    '#type' => 'submit',
+    '#title_display' => 'invisible',
+    '#value' => t('Delete'),
+    '#parents' => array('taxonomy_filters', $key, 'filter_delete_' . $key),
+    '#submit' => array('ding_nodelist_filter_delete_submit'),
+    '#limit_validation_errors' => array(),
+    '#ajax' => array(
+      'wrapper' => 'ding_nodelist-taxonomy-filters',
+      'callback' => 'ding_nodelist_js_add_filter',
+      'method' => 'replace',
+      'effect' => 'fade',
+    ),
+  );
+
+  $form['term_reference'] = array(
+    '#type' => 'value',
+    '#value' => $filter['term_reference'],
+  );
+  $form['filter_name'] = array(
```

Comment:
As mentioned previous please document the purpose of these value elements and their relationship with the other form elements.